src/xclim/sdba/base.py

```diff
@@ -195,10 +195,8 @@ def get_coordinate(self, ds: xr.Dataset | None = None) -> xr.DataArray:
         if self.prop == "dayofyear":
             if ds is not None:
                 cal = get_calendar(ds, dim=self.dim)
-                mdoy = max(
-                    xr.coding.calendar_ops._days_in_year(yr, cal)
-                    for yr in np.unique(ds[self.dim].dt.year)
-                )
+                # TODO : Change this to `ds[self.dim].dt.days_in_year.max().item()` when minimum xarray is 2024.09
+                mdoy = max_doy[cal]
```

Comment:
I imagine that this change will raise `DeprecationWarnings` until then?

Reply:
It thought it was not needed, because the output will be the same in 99% of cases. The only case where the simple `max_doy[cal]` method would be different from the fancy xarray line would be if `ds.time` only covers non-leap years. 

I want to get rid of `max_doy` and other calendar stuff when the recent additions in xarray reach the minimum SPEC0 version, but we have to wait at least one year for that...